### ISSUE TITLE
hflip is slow

### ISSUE THREAD

**Issue by vmoens** [kornia/kornia#2139]
### Describe the bug

hflip implementation is slower than it should be.
See https://github.com/kornia/kornia/blob/c552415c325980e6a75eaf8862a6350e3cd21cc2/kornia/geometry/transform/flips.py#L115-L129

I'm happy to open a PR if it's relevant.

### Reproduction steps

```bash
1. Run the following code:
python -c """                          
import torch;import timeit
x = torch.randn(3, 244, 244)
print(timeit.timeit('x.flip(-1).contiguous()', globals={'x': x}, number=10_000))
print(timeit.timeit('x[..., torch.arange(x.shape[-1] - 1, -1, -1)]', globals={'x': x, 'torch': torch}, number=10_000))
"""
On my machine I get
0.4162677849999996
1.8668735620000003
```

### Expected behavior

hflip should simply call `flip` along the last dim.

### Environment

```shell
Latest pytorch-nightly.
```

**Comment by edgarriba** [comment]
Yes, please go ahead

**Comment by shijianjian** [comment]
closed by #2179
